### PR TITLE
Update links in documentation

### LINKED ISSUE #5830: Links for spyder in documentation are broken
### Question

The links for spyder, in the documentation, seem to be broken.

For example, in [Editor and IDE Integration](https://pylint.pycqa.org/en/latest/user_guide/ide-integration.html#ide-integration) it lists a good link as [this](https://docs.spyder-ide.org/pylint.html), which gives you a 404 error.  

Also the [Pylint home page](https://www.pylint.org/) lists [this](https://pythonhosted.org/spyder/pylint.html) which can't be reached.

I'm not sure, but there are probably other places you've got the wrong spyder link...

### Documentation for future user

I think the better link is [this](https://docs.spyder-ide.org/current/panes/pylint.html)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Write a good description on what the PR does.
- [x] If you used multiple emails or multiple names when contributing, add your mails
   and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :scroll: Docs          |

## Description

A lot of stuff in that make file is unnecessary I believe. I only kept the bare minimum we need to remove some clutter.

I have also turned on `linkcheck`. Turns out a lot of our links were broken. I have fixed all dead links (or removed them), while also changing a lot of the links that were automatically redirected.

Closes #5830

### CODE REVIEW

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Would you consider making me a maintainer on RTD? I think we need to activate `linkcheck` there as well.

File: doc/Makefile [hunk 1]
```diff
@@ -11,153 +11,29 @@ PYTHON        = /usr/bin/env python
...
-	-rm -rf $(BUILDDIR)/*
-	-rm -f reference_guide/features.rst
-	-rm -f extensions.rst
-
-html: features.rst extensions.rst
+html:
 	@echo
 	@echo "Install dependencies"
-	$(PIP) install python-docs-theme
+	$(PIP) install furo
```
Comment (Pierre-Sassoulas) [inline doc/Makefile:26]: Need to install from the requirement file instead.

File: doc/Makefile [hunk 2]
```diff
@@ -11,153 +11,29 @@ PYTHON        = /usr/bin/env python
...
-
-gen-examples:
-	chmod u+w ../examples/pylintrc
-	pylint --rcfile=/dev/null --generate-rcfile > ../examples/pylintrc
-
-gen-man:
-	chmod u+w ../man/pylint.1
-	pylint --rcfile=/dev/null --generate-man > ../man/pylint.1
-
-all: html gen-man gen-examples
```
Comment (Pierre-Sassoulas) [inline doc/Makefile:163]: I don't think we should remove automatically generated option from the sphinx makefile, it cost nothing to maintain and if someone want to generate says an epub or json, it costs a lot to reinstate.

File: doc/make.bat [hunk 3]
```diff
@@ -186,13 +40,4 @@ or in %BUILDDIR%/linkcheck/output.txt.
...
 
-if "%1" == "doctest" (
-	%SPHINXBUILD% -b doctest %ALLSPHINXOPTS% %BUILDDIR%/doctest
-	if errorlevel 1 exit /b 1
-	echo.
-	echo.Testing of doctests in the sources finished, look at the ^
-results in %BUILDDIR%/doctest/output.txt.
-	goto end
-)
-
```
Comment (Pierre-Sassoulas) [inline doc/make.bat:197]: Ditto.

File: doc/whatsnew/1.7.rst [hunk 4]
```diff
@@ -444,9 +444,9 @@ New checkers
   One of the major undertakings with Python 3 was a reorganization of the standard library to
   remove old or supplanted modules and reorganize some of the existing modules.  As a result,
   roughly 100 modules that exist in Python 2 no longer exist in Python 3.  See
-  https://www.python.org/dev/peps/pep-3108/ and https://www.python.org/dev/peps/pep-0004/ for more
+  https://peps.python.org/pep-3108/ and https://peps.python.org/pep-0004/ for more
   information.  For suggestions on how to handle this, see
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/1.7.rst:448]:
```suggestion
  information.  There were suggestions on how to handle this, at
```

File: pylint/checkers/stdlib.py [hunk 5]
```diff
@@ -420,7 +420,7 @@ class StdlibChecker(DeprecatedMixin, BaseChecker):
             "unspecified-encoding",
             "It is better to specify an encoding when opening documents. "
             "Using the system default implicitly can create problems on other operating systems. "
-            "See https://www.python.org/dev/peps/pep-0597/",
+            "See hthttps://peps.python.org/pep-0597/",
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/stdlib.py:423]:
```suggestion
            "See https://peps.python.org/pep-0597/",
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

File: doc/Makefile (hunk 1 above)
Comment (DanielNoord) [reply doc/Makefile:44]: Yes, but that doesn't really work here. This should be run from within the `doc` directory so the `-e .` line in the requirements file doesn't really work.
On the other hand RTD seems to install the requirements from the root directory so for them `-e .` does work. Can't use the requirements files here.

Either way, a reference to the previous theme should probably be removed.

File: doc/Makefile (hunk 2 above)
Comment (DanielNoord) [reply doc/Makefile:163]: Re-instated 👍

**Comment by Pierre-Sassoulas** [comment]
@DanielNoord what is your read the doc username ?

**Comment by DanielNoord** [comment]
> @DanielNoord what is your read the doc username ?

DanielNoord

File: doc/Makefile (hunk 1 above)
Comment (Pierre-Sassoulas) [reply doc/Makefile:44]: Ok, not worth the headache to change the -e 😄

**Comment by DanielNoord** [comment]
Going to merge the change to links and do the update to the configuration in another PR as I'm struggling to get that right... 😅

**Pull request merged**